gtdynamics/optimizer/Optimizer.cpp

```diff
@@ -21,9 +23,41 @@ using gtsam::Values;
 
 Values Optimizer::optimize(const NonlinearFactorGraph& graph,
                            const Values& initial_values) const {
+  std::cout<< "running LM\n";
   gtsam::LevenbergMarquardtOptimizer optimizer(graph, initial_values,
                                                p_->lm_parameters);
   const Values result = optimizer.optimize();
+  std::cout << "optimization done\n";
   return result;
 }
+
+Values Optimizer::optimize(const gtsam::NonlinearFactorGraph& graph,
+                        const EqualityConstraints& constraints,
+                        const gtsam::Values& initial_values) const {
+
+  if (p_->method == OptimizationParameters::Method::UNCONSTRAINED) {
+    auto merit_graph = graph;
+    for (const auto& constraint: constraints) {
+      merit_graph.add(constraint->createFactor(1.0));
+    }
+    std::cout << "constructed merit graph\n";
+    return optimize(merit_graph, initial_values);
+  }
+  else if (p_->method == OptimizationParameters::Method::PENALTY) {
+    auto params = boost::make_shared<PenaltyMethodParameters>();
```

Comment:
this still draft I guess.. Parameters need to be passed into constructor

Reply:
Done